lib/main.js

```diff
@@ -37,6 +37,23 @@ function (moment, Router, L, GUI, numeral) {
           }
       }
 
+      var badNodes = []
```

Comment:
I can do that, though I still believe this makes some sense here.

Shouldn't a programm protect itself against faulty data and perform some checks on the data it receives? At least on the vital parts, so that it does not crash in case it receives some faulty information?